test/libsolidity/util/BytesUtils.cpp

```diff
@@ -258,7 +261,7 @@ string BytesUtils::formatBytes(
 		os << formatBoolean(_bytes);
 		break;
 	case ABIType::Hex:
-		os << formatHex(_bytes);
+		os << formatHex(_bytes, _abiType.alignDeclared);
```

Comment:
@erak Isn't `_abiType.alignDeclared` set only for expected results while parsing them? So if the expected results mistakenly don't have `left(..)`, the obtained results won't have `left(..)` either.

If I change the expected result in the test [constant_access.sol](https://github.com/ethereum/solidity/blob/185a736e13dfc8774c0ae206bddfee9a9c89a506/test/libsolidity/semanticTests/inlineAssembly/constant_access.sol) to:
```
  Expected result:
  // f() -> 2, 0xabcd, left(0x616263), true, 0x1212121212121212121212121212121212121212
```
The obtained result will be:
```
  Obtained result:
  // f() -> 2, 0xabcd000000000000000000000000000000000000000000000000000000000000, left(0x616263), true, 0x1212121212121212121212121212121212121212
```

Shouldn't left alignment be detected in this case?

Reply:
I think the test brought up in https://github.com/ethereum/solidity/issues/7028 was changed since reporting. The current expectations do look like this:

```
Expected result:
  // f() -> 2, left(0xabcd), left(0x616263), true, 0x1212...
```

Change the second parameter to `left(0xabcc)` will result in the output you've mentioned:

```
Obtained result:
  // f() -> 2, 0xabcd00..00, left(0x616263), true, 0x1212...
```
The goal was to align and shorten (but only if alignment was declared).

```
Obtained result:
  // f() -> 2, left(0xabcc), left(0x616263), true, 0x1212...
```
The behaviour about should now be implemented by this PR. Your proposal to always apply alignment if possible could be next step. Does that make sense to you?

Reply:
>The goal was to align and shorten (but only if alignment was declared).

I see what you mean. This PR seems to have fixed the issue as you've described it. Thanks for the clarification.